etcdctl/ctlv3/command/make_mirror_command.go

```diff
@@ -68,15 +72,27 @@ func makeMirrorCommandFunc(cmd *cobra.Command, args []string) {
 	dialTimeout := dialTimeoutFromCmd(cmd)
 	keepAliveTime := keepAliveTimeFromCmd(cmd)
 	keepAliveTimeout := keepAliveTimeoutFromCmd(cmd)
+
 	sec := &secureCfg{
 		cert:              mmcert,
 		key:               mmkey,
 		cacert:            mmcacert,
 		insecureTransport: mminsecureTr,
 	}
 
+	var eps []string
+	if mmusediscoverysrv {
+		srvs, err := srv.GetClient("etcd-client", args[0], "")
+		if err != nil {
+			ExitWithError(ExitError, err)
+		}
```

Comment:
what if SRV endpoint is secure?

Reply:
Doesn't `pkg/srv` take care of the http[s] discovery automatically?
https://github.com/etcd-io/etcd/blob/fa521f4e00fedfb6d98449d92a6408d0b3b0d922/pkg/srv/srv.go#L118-L119

Reply:
Aha your right my bad